td-payload/src/mm/shared.rs

```diff
@@ -9,31 +9,52 @@ use super::SIZE_4K;
 use crate::arch::shared::decrypt;
 
 static SHARED_MEMORY_ALLOCATOR: LockedHeap = LockedHeap::empty();
+static PRIVATE_SHADOW_ALLOCATOR: LockedHeap = LockedHeap::empty();
 
 pub fn init_shared_memory(start: u64, size: usize) {
+    let shadow_size = size / 2;
+    let shared_start = start + shadow_size as u64;
+    let shared_size = size - shadow_size;
+
     // Set the shared memory region to be shared
-    decrypt(start, size);
+    decrypt(shared_start, shared_size);
     // Initialize the shared memory allocator
     unsafe {
-        SHARED_MEMORY_ALLOCATOR.lock().init(start as *mut u8, size);
+        SHARED_MEMORY_ALLOCATOR
+            .lock()
+            .init(shared_start as *mut u8, shared_size);
```

Comment:
If we use double allocators, then we assume that the allocator is deterministic, not randomized. Right?

I am not sure why we need 2nd allocator, why not just use "address + offset" for the shadow?

Reply:
Got it, will follow this way